cmd/klotho/main.go

```diff
@@ -264,8 +265,12 @@ func run(cmd *cobra.Command, args []string) (err error) {
 	analyticsClient.Info("klotho compiling")
 
 	result, err := compiler.Compile(input)
-	if err != nil {
-		errHandler.PrintErr(err)
+	if err != nil || hadErrors.Load() {
+		if err != nil {
+			errHandler.PrintErr(err)
+		} else {
+			err = errors.New("Failed run of klotho invocation")
```

Comment:
I feel like this should go similar to the hadWarnings check (line 94)

Reply:
that results in a double print as well though since we already log it. i think if we are treating error log lines and errors the same, having them go through the same flow is fine, it then treats zap errors as actual errors

Reply:
we would also still print the output then if we did it that way which i thought this task is trying to prevent

Reply:
True, we do want to prevent writing the output. I guess this error wouldn't get printed since we don't explicitly print it (like we do for `err != nil`) and also set SilenceErrors. 👍